code/modules/vehicles/sealed.dm

```diff
@@ -31,7 +31,9 @@
 	. = ..()
 	if(istype(A, /obj/machinery/door))
 		var/obj/machinery/door/conditionalwall = A
-		for(var/occupant in occupants)
+		for(var/mob/occupant as anything in return_drivers())
+			if(conditionalwall.try_safety_unlock(occupant))
+				return
 			conditionalwall.bumpopen(occupant)
```

Comment:
I don't see any references to the captain being able to add accesses in the code. That said, ADDING_ACCESS_POSSIBLE & ADDING_MAINT_ACCESS_POSSIBLE are currently used to allow you to restrict who can **enter & modify the mech itself**, meaning only people with botany access can climb into it, or only dudes with crematorium access can open the maintenance panel.

The CURRENT behavior is that the ID of the pilot is being checked in bumpopen's chain. MMIs didn't have anything set up for them in the access list because... well, they don't typically move around on their own, and frankly this is kind of an underused feature.

I can easily switch it over to having to manually swipe an ID card on the MMI itself to add access levels - though it seems a bit semantic considering they could have just as easily been added to a cyborg shell that DOES have full access.

Reply:
What I'm saying is that the code makes it out to believe that updating ID *is* a thing and is probably why they made an issue report. They likely updated the access, not knowing that the access is for maintenance/access to enter, rather than the mech's access to enter places.

Reply:
Ah. I'll reword the relevant feedback to clarify that it's for access **restrictions** then.